az-hadoop-jobtype-plugin/src/main/java/azkaban/jobtype/HadoopSecureWrapperUtils.java

```diff
@@ -76,7 +76,12 @@ private static UserGroupInformation createSecurityEnabledProxyUser(
       log.info(String.format("Token = %s, %s, %s ", token.getKind(), token.getService(),
           Arrays.toString(token.getIdentifier())));
     }
-    proxyUser.addCredentials(loginUser.getCredentials());
```

Comment:
Suspecting this is the cause of the bug

Reply:
Could you please elaborate on this a bit, how was this leading to the bug ?

Reply:
Suspecting `.addCredentials()` method overwrite the tokens loaded into the UGI in the above for loop, thus causing RM delegation tokens missing

Reply:
Have you verified this? It seems like it is just adding the same credentials albeit I haven't deep-dived into the code implementation.

One way of verifying this is by printing the token state before & after removing this line in the case where cancelling fails.

Reply:
@tsangz2013  please log the token content to verify this.